### ISSUE TITLE
Error, likely from a lack of input

### ISSUE THREAD

**Issue by oconnellamethyst** [acmumn/class-lister#1]
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:30 Warning: Couldn't parse  (online)
> 2018/09/09 11:55:32 Warning: Couldn't parse  
> 2018/09/09 11:55:32 Warning: Couldn't parse  
> 2018/09/09 11:55:32 Warning: Couldn't parse  
> panic: runtime error: index out of range
> 
> goroutine 1 [running]:
> main.getCourses.func1(0x2, 0xc420527bf0)
> 	/home/travis/gopath/src/github.com/acmumn/class-lister/scraper.go:70 +0x687
> github.com/acmumn/class-lister/vendor/github.com/PuerkitoBio/goquery.(*Selection).Each(0xc420527b00, 0xc420053db8, 0x2a)
> 	/home/travis/gopath/src/github.com/acmumn/class-lister/vendor/github.com/PuerkitoBio/goquery/iteration.go:10 +0x4e
> main.getCourses(0xc420014550, 0x3, 0x3, 0x3, 0xc42019a500)
> 	/home/travis/gopath/src/github.com/acmumn/class-lister/scraper.go:34 +0x15b
> main.main()
> 	/home/travis/gopath/src/github.com/acmumn/class-lister/main.go:25 +0x3fb

If I plug in this file 
[0file.txt](https://github.com/acmumn/class-lister/files/2364446/0file.txt)
I get this error. I also get this error with a longer file. I think it's just from these line of code (marked with comments //). The code is like, I'm gonna find the thing, and the site is like there's nothing! We aren't doing any lectures! and so the code is like "THEN PERISH" at least, that's my hypothesis, which I can't test because I don't know how to code in this language. If my hypothesis is correct, commenting out the lines marked by comments and doing the go equivalent of "pass" in Python would make it work.

```golang
			if timeLocData == nil {
				// log.Println("Warning: Couldn't parse ", timeLocStr)
				return
			}
			startTime := timeLocData[1]
			days := timeLocData[2]
			room := timeLocData[3]
			if room == "(online)" {
				return
			}
			instructor := unfuck(e.Find(":nth-child(4)"))

			c := &courses[len(courses)-1]
			c.Lectures = append(c.Lectures, Lecture{
				Time:      startTime,
				Days:      days,
				Room:      room,
				Professor: instructor,
			})
		} else {
			// panic("TODO")
		}
	})
	return courses
```
But I don't actually know how to actually make those changes because I don't know how to Go. I attempted to make the changes on my fork and got stuck trying to make a .bin

**Comment by oconnellamethyst** [comment]
It could also be that I'm doing too many inputs though

**Comment by oconnellamethyst** [comment]
Trying to figure that out right now

**Comment by remexre** [comment]
Looking at what it's trying and failing to parse, an assumption I made earlier in development (the department offers at least one undergraduate course in the current term) isn't true for all the departments you're trying, e.g. [`ACL`](https://www.aem.umn.edu/cgi-bin/courses/noauth/class-schedule?current=ACL).

Also, it looks like business admin has a class with a number <1000 (`BA 999`)... Also what the heck even is that department, they have a course literally named "Career Skills."

Anyway, this should be fixed by 61b4600.
